### PR TITLE
Fix read timeout and add default timeout values.

### PR DESCRIPTION
**Pull request by ericdreeves**
By setting the read deadline in streamReader.Read(), the deadline was
extended by the read timeout on each subsequent call. To avoid this, the
deadline is set in FCGIClient.Request(), before the first read occurs.

Additionally, the read_timeout and connect_timeout properties now have a
default value of 60 seconds.

See #1094.

### CODE REVIEW

**Comment by ericdreeves** [comment]
As I look over this and start on the write_timeout property I'm thinking we might extract the rwc field on FCGIClient into a (unexported?) type that implements the net.Conn interface.

That would simplify things a bit. Thoughts?

**Comment by mholt** [comment]
Thanks for taking initiative on this, @ericdreeves ! I don't have a chance to look at it yet but maybe somebody else does?

File: caddyhttp/fastcgi/setup.go [hunk 1]
```diff
@@ -53,15 +53,13 @@ func fastcgiParse(c *caddy.Controller) ([]Rule, error) {
...
-		var rule Rule
-
 		args := c.RemainingArgs()
 
 		if len(args) < 2 || len(args) > 3 {
 			return rules, c.ArgErr()
 		}
 
-		rule.Path = args[0]
+		rule := Rule{Path: args[0], ReadTimeout: 60 * time.Second}
```
Comment (wendigo) [inline caddyhttp/fastcgi/setup.go]: I know that 60 seconds is a lot of time but this change can actually break existing configurations. I'm not really sure what value should be used as default. Suggestions @mholt ?

Comment (mholt) [inline caddyhttp/fastcgi/setup.go]: Why can this break existing configs?

Comment (ericdreeves) [inline caddyhttp/fastcgi/setup.go]: Thanks for pointing it out, @wendigo.

@mholt some configurations may rely on a read timeout of greater than 60 seconds, whether users know it or not. Affected users would have to update their Caddyfile with the new read_timeout property in order to preserve the previous behavior. And even if the property is used, the exact duration might not be known.

Some options:
1. I can remove the default value and users must explicitly declare the read_timeout
2. Make the default duration greater than the currently proposed one

Are there others?

Comment (wendigo) [inline caddyhttp/fastcgi/setup.go]: @mholt If it's explicitly stated in the documentation - I'm fine with default value 👍

Comment (ericdreeves) [inline caddyhttp/fastcgi/setup.go]: Thanks for the feedback.

Speaking of documentation, I'm happy to make those changes as well.

**Comment by mholt** [comment]
Great, thank you for making progress on this!! :tada:

**Pull request merged**